pkg/streamingpromql/planning.go

```diff
@@ -494,6 +495,31 @@ func (p *QueryPlanner) nodeFromExpr(expr parser.Expr) (planning.Node, error) {
 			return nil, compat.NewNotSupportedError(fmt.Sprintf("'%s' function", expr.Func.Name))
 		}
 
+		if fnc == functions.FUNCTION_INFO {
+			if len(expr.Args) == 1 {
+				infoExpr, err := parser.ParseExpr("target_info")
+				if err != nil {
+					return nil, err
+				}
+				expr.Args = append(expr.Args, infoExpr)
```

Comment:
It might be good to do this as a very early AST optimisation pass. This will allow all other optimisation passes to ignore the fact that `info` can have one or two arguments, and instead they can just handle the two argument case.